src/FffGcodeWriter.cpp

```diff
@@ -1640,7 +1641,78 @@ bool FffGcodeWriter::addSupportToGCode(const SliceDataStorage& storage, LayerPla
     return support_added;
 }
 
-bool FffGcodeWriter::addSupportInfillToGCode(const SliceDataStorage& storage, LayerPlan& gcode_layer, int layer_nr) const
+
+bool FffGcodeWriter::processMultiLayerSupportInfill(const SliceDataStorage& storage, LayerPlan& gcode_layer, unsigned int layer_nr, const int extruder_nr) const
+{
+    if (extruder_nr != storage.getSettingAsIndex("support_infill_extruder_nr"))
+    {
+        return false;
+    }
+    bool added_something = false;
+
+    int64_t z = layer_nr * getSettingInMicrons("layer_height");
+
+    const ExtruderTrain& infill_extr = *storage.meshgroup->getExtruderTrain(getSettingAsIndex("support_infill_extruder_nr"));
+    int support_infill_line_distance = infill_extr.getSettingInMicrons("support_line_distance");
+    int support_infill_overlap = 0; // support infill should not be expanded outward
+    EFillMethod support_pattern = storage.meshgroup->getSettingAsFillMethod("support_pattern");
+    if (support_pattern == EFillMethod::GRID || support_pattern == EFillMethod::TRIANGLES || support_pattern == EFillMethod::CONCENTRIC)
+    {
+        support_infill_overlap = infill_extr.getSettingInMicrons("infill_overlap_mm"); // support lines area should be expanded outward to overlap with the boundary polygon
+    }
+    double support_infill_angle = 0;
+
+    if (support_infill_line_distance <= 0)
+    {
+        return false;
+    }
+
+    //Print the thicker infill lines first. (double or more layer thickness, infill combined with previous layers)
+    const std::vector<SupportInfillPart>& part_list = storage.support.supportLayers[layer_nr].support_infill_parts;
+    for (unsigned int part_idx = 0; part_idx < part_list.size(); ++part_idx)
```

Comment:
This function is getting pretty big.

I suggest you split up this function here and introduce a new function to process the multilayer support of one support infill part.
The function is then similar to the function processMultiLayerInfill which works on a single SliceLayerPart.

[not blocking]